R/estimateAlpha.R

```diff
@@ -0,0 +1,245 @@
+            FUN <- .estimate_richness
+        } else {
+            stop("'index' is coresponding to none of the alpha diversity indices.
+                  'index' should be one of: coverage_diversity, fisher_diversity,
+                  faith_diversity, gini_simpson_diversity,
+                  inverse_simpson_diversity, log_modulo_skewness_diversity,
+                  shannon_diversity, absolute_dominance, dbp_dominance,
+                  core_abundance_dominance, gini_dominance,
+                  dmn_dominance, relative_dominance, simpson_lambda_dominance,
+                  camargo_evenness, pielou_evenness, simpson_evenness,
+                  evar_evenness, bulla_evenness, ace_richness, chao1_richness,
+                  hill_richness or observed_richness.",
+                 call. = FALSE)
+        }
+        # Performing rarefaction if rarefaction.depth is specified to be less 
+        # the max of the total read counts
+        if (rarefaction.depth < max(colSums(assay(x, assay.type)), na.rm = TRUE)) {
+            x <- .alpha_rarefaction(x, n.iter = n.iter,
+                                    args.sub = list(assay.type=assay.type,
+                                                    min_size=rarefaction.depth,
+                                                    verbose=FALSE),
+                                    FUN=FUN,
+                                    args.fun=list(index=index[i], assay.type="subsampled"),
+                                    ...,
+                                    name=name[i])
+        } else {
+            # Estimate index without rarefaction
+            suppressWarnings(x <- do.call(FUN, args = c(list(x, assay.type=assay.type,
+                                                             index=index[i],
+                                                             name=name[i]),
+                                                        list(...))))
+        }
+    }
+    return(x)
+}
+
+## Helper functions
+
+.get_indices <- function(measure) {
+    switch(measure,
+           "diversity" = c("coverage_diversity", "faith_diversity", 
+                           "fisher_diversity", "gini_simpson_diversity",
+                           "inverse_simpson_diversity",
+                           "log_modulo_skewness_diversity", "shannon_diversity"),
+           "dominance" = c("absolute_dominance",
+                           "dbp_dominance", "core_abundance_dominance",
+                           "gini_dominance", "dmn_dominance", "relative_dominance",
+                           "simpson_lambda_dominance"),
+           "evenness" = c("camargo_evenness", "pielou_evenness", "simpson_evenness",
+                          "evar_evenness", "bulla_evenness"),
+           "richness" = c("ace_richness", "chao1_richness", "hill_richness", 
+                          "observed_richness"))
+}
+
+.alpha_rarefaction <- function(x,
+                               n.iter=1L,
```

Comment:
As a note, you could add line break after `function(` --> the code would look nicer